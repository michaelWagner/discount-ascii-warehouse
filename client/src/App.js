import React, { Component } from 'react';
import ProductGrid from './ProductGrid';
import Product from './Product';

class App extends Component {
  state = {
    products: [],
    cart: []
  }

  removeProductItem = (itemIndex) => {
    const filteredProducts = this.state.products.filter(
      (item, idx) => itemIndex !== idx,
    );
    this.setState({ products: filteredProducts });
  }

  addProductToCart = (product) => {
    const newCart = this.state.cart.concat(product);
    this.setState({ cart: newCart });
  }

  viewProduct = (product) => {
    // const newProducts = this.state.products.concat(product);
    // this.setState({ products: newProducts });
  }

  render() {
    const { products, cart } = this.state;

    return (
      <div className='App'>
        <div className='ui text container'>
        <ProductGrid />
        </div>
      </div>
    );
  }
}

export default App;
