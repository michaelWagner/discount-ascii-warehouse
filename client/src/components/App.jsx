import React, { Component } from 'react';
import InfiniteScroll from 'react-infinite-scroller';
import ProductGrid from './ProductGrid';
import Loading from './Loading';
import Advertisement from './Advertisement';

class App extends Component {
  constructor() {
    super();
    this.state = {
      hasProductGridLoaded: false,
      componentName: 'App',
      randomAd: null,
      adList: []
    };
  }

  addAdtoAdList(ad) {
    let newAdList = this.state.adList.concat(ad);
    this.setState({randomAd: ad, adList: newAdList });
  }

  // loadMore() {
  //   ProductGrid.loadProducts('limit=400');
  // }

  generateRandomAd() {
    let random = Math.floor(Math.random() * 1000);
    if (this.state.adList.length > 1) {
      while (random === this.state.adList[-1]) {
        random = Math.floor(Math.random() * 1000);
      }
    }
    this.addAdtoAdList(random)
    return random;
  }

  hasLoaded() {
    this.setState({hasProductGridLoaded: true});
  }

  render() {
    const loadState = (
      this.state.hasProductGridLoaded
      ? "hide-loading-bar"
      : "loading-bar"
    );

    return (
      <div className='App'>
        <header>
          <h1>Discount Ascii Warehouse</h1>

          <p className="tag-line">
            Here you're sure to find a bargain on some of the finest ascii
            available to purchase. Be sure to peruse our selection of ascii
            faces in an exciting range of sizes and prices.
          </p>

        </header>
        <p className="ad-tag-line">But first, a word from our sponsors:</p>
        <Advertisement key={0} adList={this.state.adList} randomAd={this.randomAd} generateRandomAd={this.generateRandomAd.bind(this)} componentName={this.state.componentName}/>

        <Loading hasLoaded={loadState} />
        <ProductGrid adList={this.state.adList} randomAd={this.randomAd} generateRandomAd={this.generateRandomAd.bind(this)} hasProductGridLoaded={this.hasLoaded.bind(this)}/>
      </div>
    );
  }
}

export default App;
