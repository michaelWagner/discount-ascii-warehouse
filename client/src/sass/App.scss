$discount-ascii-warehouse-color: #4CAF50;

body {
  width: 100%;
  margin: 0;
  padding: 0;
}

.App {
  text-align: left;
  header {
    h1 {
      background-color: $discount-ascii-warehouse-color;
      width: 100%;
      padding: 14px 0;
      margin: 0;
      text-align: center;
      color: #fff;
    }
  }

  .loader-indicator.fade-enter,
  .loader-content.fade-enter {
    opacity: 0;
  }

  .loader-indicator.fade-enter-active,
  .loader-content.fade-enter-active {
    opacity: 1;
  }

  .loader-indicator.fade-leave,
  .loader-content.fade-leave {
    opacity: 1;
  }

  .loader-indicator.fade-leave-active,
  .loader-content.fade-leave-active {
    opacity: 0;
  }

  .loader-content,
  .loader-indicator {
    transition: opacity 1s;
  }

  .loader-ellipsisDot {
    padding: 0 3px 0 1px;
    opacity: 0;
    animation-duration: 1300ms;
    animation-iteration-count: infinite;
    animation-timing-function: linear;
  }

  @keyframes loader-ellipsisDot1 {
    0%   { opacity: 0; }
    25%  { opacity: 0; }
    35%  { opacity: 1; }
    100% { opacity: 1; }
  }

    @keyframes loader-ellipsisDot2 {
    0%   { opacity: 0; }
    50%  { opacity: 0; }
    60%  { opacity: 1; }
    100% { opacity: 1; }
  }

  @keyframes loader-ellipsisDot3 {
    0%   { opacity: 0; }
    75%  { opacity: 0; }
    85%  { opacity: 1; }
    100% { opacity: 1; }
  }

  .loader-ellipsisDot:nth-child(1) {
    animation-name: loader-ellipsisDot1;
  }

  .loader-ellipsisDot:nth-child(2) {
    animation-name: loader-ellipsisDot2;
  }

  .loader-ellipsisDot:nth-child(3) {
    animation-name: loader-ellipsisDot3;
  }
}

.ad {
}

.loading-bar {
  display: block;
}

.hide-loading-bar {
  display: none;
}

#product-grid {
  width: 95%;
  margin: auto;

  table {
    border-collapse: collapse;

    th, td {
      border: 1px solid black;
      padding: 10px;
      cursor: pointer;
    }
    th {
      background-color: $discount-ascii-warehouse-color;
      color: white;
    }
    tr:nth-child(even) {background-color: #f2f2f2}
  }
}
