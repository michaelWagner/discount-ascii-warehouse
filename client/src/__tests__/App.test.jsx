import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import App from '../components/App';

describe(App, () => {

  const component = shallow(
    <App />
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <App />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('doesn\'t load ProductGrid initially', () => {
    expect(component.state('hasProductGridLoaded')).toBeFalsy();
  });
});
