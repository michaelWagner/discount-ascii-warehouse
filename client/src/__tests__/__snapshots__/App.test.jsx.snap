exports[`class App extends /* istanbul ignore next */_react.Component {
  constructor() {
    super();

    this.state = {
      hasProductGridLoaded: false,
      cartVisible: false,
      cart: [],
      adList: [] };


    this.defaultProps = { componentName: 'App' };
  }

  addProductToCart(product) {
    const newCart = this.state.cart.concat(product);
    this.setState({ cart: newCart }, () => {
    });
  }

  formatPriceFromCentsToDollars(price) {
    if (price) {
      // Convert price to Dollar and cents notation in the format of '$1.50'.
      return '$' + (parseInt(price, 10) / 100).toFixed(2);
    } else {
      return '$0.00';
    }
  }

  generateRandomId() {
    // Make sure randomId is unique.
    let randomId = Math.floor(Math.random() * 1000);

    while (this.state.adList.indexOf(randomId) > 0) {
      randomId = Math.floor(Math.random() * 1000);
    }

    const newAdList = this.state.adList.concat(randomId);
    this.setState({ adList: newAdList });

    return randomId;
  }

  hasLoaded() {
    this.setState({ hasProductGridLoaded: true });
    return true;
  }

  removeProductFromCart(product) {
    const newCart = this.state.cart.filter((item, idx) => item.id !== product.id);
    this.setState({ cart: newCart });
  }

  toggleCartVisiblity() {
    const cartVisible = this.state.cartVisible ? false : true;
    this.setState({ cartVisible: cartVisible });
  }

  render() {
    const loadState =
    this.state.hasProductGridLoaded ?
    "hide-loading-bar" :
    "loading-bar";


    const totalPrice = this.state.cart.reduce((total, product) => {
      return total + product.price;
    }, 0);

    return (
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'App', __source: { fileName: _jsxFileName, lineNumber: 77 }, __self: this },
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'header', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 78 }, __self: this },
          /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'h1', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 79 }, __self: this }, /* istanbul ignore next */'Discount Ascii Warehouse'),
          /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'cart-btn-wrapper', onClick: this.toggleCartVisiblity.bind(this), __source: { fileName: _jsxFileName, lineNumber: 80 }, __self: this },
            /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'cart-total-price', __source: { fileName: _jsxFileName, lineNumber: 81 }, __self: this }, this.formatPriceFromCentsToDollars(totalPrice)),
            /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'cart-btn', __source: { fileName: _jsxFileName, lineNumber: 82 }, __self: this }))),


        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Cart2.default, /* istanbul ignore next */{ products: this.state.cart,
          totalPrice: totalPrice,
          toggleProductInCart: this.removeProductFromCart.bind(this),
          formatPriceFromCentsToDollars: this.formatPriceFromCentsToDollars,
          cartVisible: this.state.cartVisible, __source: { fileName: _jsxFileName, lineNumber: 85 }, __self: this }),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'key-visual-wrapper', __source: { fileName: _jsxFileName, lineNumber: 90 }, __self: this },
          /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'key-visual', __source: { fileName: _jsxFileName, lineNumber: 91 }, __self: this },
            /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'tag-line-wrapper', __source: { fileName: _jsxFileName, lineNumber: 92 }, __self: this },
              /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'p', /* istanbul ignore next */{ className: 'tag-line', __source: { fileName: _jsxFileName, lineNumber: 93 }, __self: this }, /* istanbul ignore next */'Here you're sure to find a bargain on some of the finest ascii available to purchase. Be sure to peruse our selection of ascii faces in an exciting range of sizes and prices.')),





            /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'key-visual-img', __source: { fileName: _jsxFileName, lineNumber: 99 }, __self: this }))),



        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'p', /* istanbul ignore next */{ className: 'ad-tag-line', __source: { fileName: _jsxFileName, lineNumber: 103 }, __self: this }, /* istanbul ignore next */'But first, a word from our sponsors:'),

        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Advertisement2.default, /* istanbul ignore next */{ key: 0, generateRandomId: this.generateRandomId.bind(this),
          componentName: this.defaultProps.componentName, __source: { fileName: _jsxFileName, lineNumber: 105 }, __self: this }),

        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Loading2.default, /* istanbul ignore next */{ hasLoaded: loadState, __source: { fileName: _jsxFileName, lineNumber: 108 }, __self: this }),

        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'p', /* istanbul ignore next */{ className: 'product-grid-guide', __source: { fileName: _jsxFileName, lineNumber: 110 }, __self: this }, /* istanbul ignore next */'Click column header to sort. Click on product to add to cart.'),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_ProductGrid2.default, /* istanbul ignore next */{ toggleProductInCart: this.addProductToCart.bind(this),
          generateRandomId: this.generateRandomId.bind(this),
          formatPriceFromCentsToDollars: this.formatPriceFromCentsToDollars,
          hasProductGridLoaded: this.hasLoaded.bind(this), __source: { fileName: _jsxFileName, lineNumber: 111 }, __self: this })));


  }} renders and matches our snapshot 1`] = `
<div
  className="App">
  <header>
    <h1>
      Discount Ascii Warehouse
    </h1>
    <div
      className="cart-btn-wrapper"
      onClick={[Function]}>
      <div
        className="cart-total-price">
        $0.00
      </div>
      <div
        className="cart-btn" />
    </div>
  </header>
  <table
    className="cart-hidden product-grid-table">
    <tfoot
      className="cart-table-footer">
      <tr
        className="table-footer">
        <td
          colSpan="2">
          Total price: 
        </td>
        <td
          className="table-footer-price"
          colSpan={4}>
          $0.00
        </td>
      </tr>
    </tfoot>
  </table>
  <div
    className="key-visual-wrapper">
    <div
      className="key-visual">
      <div
        className="tag-line-wrapper">
        <p
          className="tag-line">
          Here you're sure to find a bargain on some of the finest ascii available to purchase. Be sure to peruse our selection of ascii faces in an exciting range of sizes and prices.
        </p>
      </div>
      <div
        className="key-visual-img" />
    </div>
  </div>
  <p
    className="ad-tag-line">
    But first, a word from our sponsors:
  </p>
  <img
    alt=""
    className="ad"
    src="/ad/?r=71" />
  <div
    className="loading-bar">
    <span
      className="loader">
      <div
        className="loader-indicator">
        <h1>
          <span>
            Loading
          </span>
          <span
            className="loader-ellipsis">
            <span
              className="loader-ellipsisDot">
              .
            </span>
            <span
              className="loader-ellipsisDot">
              .
            </span>
            <span
              className="loader-ellipsisDot">
              .
            </span>
          </span>
        </h1>
      </div>
    </span>
  </div>
  <p
    className="product-grid-guide">
    Click column header to sort. Click on product to add to cart.
  </p>
  <div
    id="product-grid" />
</div>
`;

exports[`class App extends /* istanbul ignore next */_react.Component {
  constructor() {
    super();

    this.state = {
      hasProductGridLoaded: false,
      cartVisible: false,
      cart: [],
      adList: [] };


    this.defaultProps = { componentName: 'App' };
  }

  addProductToCart(product) {
    const newCart = this.state.cart.concat(product);
    this.setState({ cart: newCart }, () => {
    });
  }

  formatPriceFromCentsToDollars(price) {
    if (price) {
      // Convert price to Dollar and cents notation in the format of '$1.50'.
      return '$' + (parseInt(price, 10) / 100).toFixed(2);
    } else {
      return '$0.00';
    }
  }

  generateRandomId() {
    // Make sure randomId is unique.
    let randomId = Math.floor(Math.random() * 1000);

    while (this.state.adList.indexOf(randomId) > 0) {
      randomId = Math.floor(Math.random() * 1000);
    }

    const newAdList = this.state.adList.concat(randomId);
    this.setState({ adList: newAdList });

    return randomId;
  }

  hasLoaded() {
    this.setState({ hasProductGridLoaded: true });
    return true;
  }

  removeProductFromCart(product) {
    const newCart = this.state.cart.filter((item, idx) => item.id !== product.id);
    this.setState({ cart: newCart });
  }

  toggleCartVisiblity() {
    const cartVisible = this.state.cartVisible ? false : true;
    this.setState({ cartVisible: cartVisible });
  }

  render() {
    const loadState =
    this.state.hasProductGridLoaded ?
    "hide-loading-bar" :
    "loading-bar";


    const totalPrice = this.state.cart.reduce((total, product) => {
      return total + product.price;
    }, 0);

    return (
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'App', __source: { fileName: _jsxFileName, lineNumber: 77 }, __self: this },
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'header', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 78 }, __self: this },
          /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'h1', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 79 }, __self: this }, /* istanbul ignore next */'Discount Ascii Warehouse'),
          /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'cart-btn-wrapper', onClick: this.toggleCartVisiblity.bind(this), __source: { fileName: _jsxFileName, lineNumber: 80 }, __self: this },
            /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'cart-total-price', __source: { fileName: _jsxFileName, lineNumber: 81 }, __self: this }, this.formatPriceFromCentsToDollars(totalPrice)),
            /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'cart-btn', __source: { fileName: _jsxFileName, lineNumber: 82 }, __self: this }))),


        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Cart2.default, /* istanbul ignore next */{ products: this.state.cart,
          totalPrice: totalPrice,
          toggleProductInCart: this.removeProductFromCart.bind(this),
          formatPriceFromCentsToDollars: this.formatPriceFromCentsToDollars,
          cartVisible: this.state.cartVisible, __source: { fileName: _jsxFileName, lineNumber: 85 }, __self: this }),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'key-visual-wrapper', __source: { fileName: _jsxFileName, lineNumber: 90 }, __self: this },
          /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'key-visual', __source: { fileName: _jsxFileName, lineNumber: 91 }, __self: this },
            /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'tag-line-wrapper', __source: { fileName: _jsxFileName, lineNumber: 92 }, __self: this },
              /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'p', /* istanbul ignore next */{ className: 'tag-line', __source: { fileName: _jsxFileName, lineNumber: 93 }, __self: this }, /* istanbul ignore next */'Here you\'re sure to find a bargain on some of the finest ascii available to purchase. Be sure to peruse our selection of ascii faces in an exciting range of sizes and prices.')),





            /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'key-visual-img', __source: { fileName: _jsxFileName, lineNumber: 99 }, __self: this }))),



        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'p', /* istanbul ignore next */{ className: 'ad-tag-line', __source: { fileName: _jsxFileName, lineNumber: 103 }, __self: this }, /* istanbul ignore next */'But first, a word from our sponsors:'),

        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Advertisement2.default, /* istanbul ignore next */{ key: 0, generateRandomId: this.generateRandomId.bind(this),
          componentName: this.defaultProps.componentName, __source: { fileName: _jsxFileName, lineNumber: 105 }, __self: this }),

        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Loading2.default, /* istanbul ignore next */{ hasLoaded: loadState, __source: { fileName: _jsxFileName, lineNumber: 108 }, __self: this }),

        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'p', /* istanbul ignore next */{ className: 'product-grid-guide', __source: { fileName: _jsxFileName, lineNumber: 110 }, __self: this }, /* istanbul ignore next */'Click column header to sort. Click on product to add to cart.'),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_ProductGrid2.default, /* istanbul ignore next */{ toggleProductInCart: this.addProductToCart.bind(this),
          generateRandomId: this.generateRandomId.bind(this),
          formatPriceFromCentsToDollars: this.formatPriceFromCentsToDollars,
          hasProductGridLoaded: this.hasLoaded.bind(this), __source: { fileName: _jsxFileName, lineNumber: 111 }, __self: this })));


  }} renders and matches our snapshot 1`] = `
<div
  className="App">
  <header>
    <h1>
      Discount Ascii Warehouse
    </h1>
    <div
      className="cart-btn-wrapper"
      onClick={[Function]}>
      <div
        className="cart-total-price">
        $0.00
      </div>
      <div
        className="cart-btn" />
    </div>
  </header>
  <table
    className="cart-hidden product-grid-table">
    <tfoot
      className="cart-table-footer">
      <tr
        className="table-footer">
        <td
          colSpan="2">
          Total price: 
        </td>
        <td
          className="table-footer-price"
          colSpan={4}>
          $0.00
        </td>
      </tr>
    </tfoot>
  </table>
  <div
    className="key-visual-wrapper">
    <div
      className="key-visual">
      <div
        className="tag-line-wrapper">
        <p
          className="tag-line">
          Here you\'re sure to find a bargain on some of the finest ascii available to purchase. Be sure to peruse our selection of ascii faces in an exciting range of sizes and prices.
        </p>
      </div>
      <div
        className="key-visual-img" />
    </div>
  </div>
  <p
    className="ad-tag-line">
    But first, a word from our sponsors:
  </p>
  <img
    alt=""
    className="ad"
    src="/ad/?r=750" />
  <div
    className="loading-bar">
    <span
      className="loader">
      <div
        className="loader-indicator">
        <h1>
          <span>
            Loading
          </span>
          <span
            className="loader-ellipsis">
            <span
              className="loader-ellipsisDot">
              .
            </span>
            <span
              className="loader-ellipsisDot">
              .
            </span>
            <span
              className="loader-ellipsisDot">
              .
            </span>
          </span>
        </h1>
      </div>
    </span>
  </div>
  <p
    className="product-grid-guide">
    Click column header to sort. Click on product to add to cart.
  </p>
  <div
    id="product-grid" />
</div>
`;
